ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -622,6 +622,12 @@ static const upb_pbdecodermethod *msgdef_decodermethod(Descriptor* desc) {
   return desc->fill_method;
 }
 
+// Callback invoked by upb if any error occurs during parsing or serialization.
+static bool env_error_func(void* ud, const upb_status* status) {
+  const char* message = (const char*)ud;
+  rb_raise(rb_eRuntimeError, message, upb_status_errmsg(status));
```

Comment:
btw, I haven't documented in upb the guarantees around longjmp-ing across an encoder or decoder. I think the semantic that will make the most sense is: you can longjmp across an encoder or decoder and all resources will be properly be freed when you call upb_env_uninit(), but it is no longer allowed to *use* the environment for anything except freeing it. Because by longjmp-ing, you may have jumped over code that is necessary for the parse state to be fully synchronized. For example, with JIT-ted code, the proper suspending logic would copy the CPU stack into the upb_pbdecoder prior to returning. By skipping over that, the decoder cannot properly be resumed.